jscomp/core/lam_compile_primitive.ml

```diff
@@ -36,8 +36,44 @@ let ensure_value_unit (st : Lam_compile_context.continuation) e : E.t =
   | EffectCall Not_tail -> e
 (* NeedValue should return a meaningful expression*)
 
-let translate loc (cxt : Lam_compile_context.t) (prim : Lam_primitive.t)
-    (args : J.expression list) : J.expression =
+let module_of_expression = function
+  | J.Var (J.Qualified (module_id, value)) -> [ (module_id, value) ]
+  | _ -> []
+
+let get_module_system () =
+    let package_info = Js_packages_state.get_packages_info () in
+    let module_system =
+        if Js_packages_info.is_empty package_info && !Js_config.js_stdout then
+            [Js_packages_info.NodeJS]
+        else Js_packages_info.map package_info (fun {module_system} -> module_system)
+    in
+    match module_system with
+    | [module_system] -> module_system
+    | _ -> NodeJS
```

Comment:
https://github.com/rescript-lang/rescript-compiler/blob/master/jscomp/core/lam_compile_main.ml#L294-L297

Not sure in what cases the module_system in Js_packages_info can be multiple in one runtime. For now, I've handled the case where there is only one.

Reply:
Perhaps we need some tests on fairly large projects.